### PR TITLE
add govulncheck scan to Makefile

### PR DESCRIPTION
**Pull request by sheharyaar**
Signed-off-by: Mohammad Shehar Yaar Tausif <sheharyaar48@gmail.com>

### CODE REVIEW

**Comment by sheharyaar** [comment]
It includes the test files via the `-test` flag

**Review by kranurag7 — CHANGES_REQUESTED** [review]
Hey @sheharyaar With the latest version of `govulncheck`, `-v` flag is not defined anymore, so we don't need that anymore. Rest looks good to me.

**Comment by sheharyaar** [comment]
Sure, will make the required changes.

**Comment by sheharyaar** [comment]
@kranurag7 it's done

**Review by kranurag7 — APPROVED** [review]
LGTM @sheharyaar, thanks for working on this.

File: Makefile [hunk 1]
```diff
@@ -67,3 +67,9 @@ ifeq (, $(shell which gosec))
 	}
 endif
 	cd $(CURDIR); gosec ./...
+
+.PHONY: scan
+scan: 
+	go install golang.org/x/vuln/cmd/govulncheck@latest ;\
```
Comment (daemon1024) [inline Makefile:73]: We should check if it exists before we install ig?

Comment (kranurag7) [reply Makefile:73]: Yes, very good catch 😃. @sheharyaar can you please add a check using `command -v govulncheck` before invoking `go install`

Comment (sheharyaar) [reply Makefile:73]: Sure

**Comment by sheharyaar** [comment]
It's done

**Review by daemon1024 — APPROVED** [review]

**Comment by daemon1024** [comment]
Will adding it to CI be a different task, @kranurag7 @sheharyaar ?

**Comment by kranurag7** [comment]
> Will adding it to CI be a different task, @kranurag7 @sheharyaar ?

Can be handled in a separate PR using this reusable action https://github.com/golang/govulncheck-action 
It also requires more work, I'm not sure if the action returns a non-zero exit code when it finds some vulnerabilities. A good thing would be to take a look at how other repos are using this reusable action.
Ref: https://github.com/search?q=path%3A%2F%5E.github%5C%2Fworkflows%2F+golang%2Fgovulncheck&type=code

**Pull request merged**